src/backend/distributed/replication/multi_logical_replication.c

```diff
@@ -1204,16 +1156,18 @@ CreateForeignConstraintsToReferenceTable(List *logicalRepTargetList)
 		 */
 		foreach_ptr(shardInterval, target->newShards)
 		{
-			List *commandList = GetForeignConstraintCommandsToReferenceTable(
+			List *commandList = CopyShardForeignConstraintCommandList(
 				shardInterval);
+			commandList = list_concat(
+				list_make1("SET LOCAL citus.skip_constraint_validation TO ON;"),
+				commandList);
 
-			char *command = NULL;
+			SendCommandListToWorkerOutsideTransaction(
+				target->superuserConnection->hostname,
+				target->superuserConnection->port,
+				target->superuserConnection->user,
+				commandList);
 
-			/* iterate over the commands and execute them in the same connection */
-			foreach_ptr(command, commandList)
-			{
-				ExecuteCriticalRemoteCommand(target->superuserConnection, command);
```

Comment:
There was actually a bug here, because the result from `GetForeignConstraintCommandsToReferenceTable` wasn't run inside a transaction. So the `SET LOCAL` that it returned wasn't actually used.